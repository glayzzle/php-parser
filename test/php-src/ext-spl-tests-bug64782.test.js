// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/spl/tests/bug64782.phpt
  it("Bug #64782: SplFileObject constructor make $context optional / give it a default value", function () {
    expect(parser.parseCode("<?php\nvar_dump(new SplFileObject(__FILE__, \"r\", false, null));\n?>")).toMatchSnapshot();
  });
});
