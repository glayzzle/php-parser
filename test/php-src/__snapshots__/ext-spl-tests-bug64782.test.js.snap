// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Bug #64782: SplFileObject constructor make $context optional / give it a default value 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          New {
            "arguments": Array [
              Magic {
                "kind": "magic",
                "raw": "__FILE__",
                "value": "__FILE__",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"r\\"",
                "unicode": false,
                "value": "r",
              },
              Boolean {
                "kind": "boolean",
                "raw": "false",
                "value": false,
              },
              NullKeyword {
                "kind": "nullkeyword",
                "raw": "null",
              },
            ],
            "kind": "new",
            "what": Name {
              "kind": "name",
              "name": "SplFileObject",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
